crates/proto/src/serialize/binary/encoder.rs

```diff
@@ -241,19 +241,18 @@ impl<'a> BinEncoder<'a> {
         assert!(end <= (u16::max_value() as usize));
         assert!(start <= end);
         if self.offset < 0x3FFF_usize {
-            self.name_pointers.push((start, end)); // the next char will be at the len() location
+            self.name_pointers.push((start, self.slice_of(start, end).to_vec())); // the next char will be at the len() location
         }
     }
 
     /// Looks up the index of an already written label
     pub fn get_label_pointer(&self, start: usize, end: usize) -> Option<u16> {
         let search = self.slice_of(start, end);
 
-        for &(match_start, match_end) in &self.name_pointers {
-            let matcher = self.slice_of(match_start as usize, match_end as usize);
-            if matcher == search {
-                assert!(match_start <= (u16::max_value() as usize));
-                return Some(match_start as u16);
+        for (match_start, matcher) in &self.name_pointers {
```

Comment:
should this go through the entire list and find the longest match? Right now it only finds the first matching name_pointer, but not the longest.

Thoughts?

Reply:
We have already started matching from the longest match.
https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/rr/domain/name.rs#L618